tasks-router.js

```diff
@@ -0,0 +1,59 @@
+const express = require("express");
+
+const Tasks = require("./data/tasksModel.js");
+
+const router = express.Router();
+
+module.exports = router;
+
+router.post("/", async (req, res) => {
+    const newTask = req.body;
+
+    if (newTask.description) {
+        try {
+            const task = await Tasks.insert(req.body);
+            res.status(201).json(task);
+        } catch (err) {
+            console.log(err);
+            res.status(500).json({
+                message:
+                    "There was an error while saving the task to the database"
+            });
+        }
+    } else {
+        res.status(400).json({
+            err: "Please provide description for the resources."
+        });
+    }
+});
+
+router.get("/", async (req, res) => {
+    try {
+        const tasks = await Tasks.get(req.query);
+        res.status(200).json(tasks);
+    } catch (err) {
+        console.log(err);
+        res.status(500).json({
+            message: "The tasks information could not be retrieved."
+        });
+    }
+});
+
+router.get("/:id", async (req, res) => {
+    try {
+        const tasks = await Tasks.getById(req.params.id);
+
+        if (post) {
+            res.status(200).json(task);
+        } else {
+            res.status(404).json({
+                message: "The task with the specified ID does not exist"
+            });
+        }
+    } catch (error) {
+        console.log(error);
+        res.status(500).json({
+            message: "The task information could not be retrieved"
+        });
+    }
+});
```

Comment:
I told you I would show you how I structure my router pages...Here is an example: 

```
const router = require('express').Router();
const Projects = require('../models/projectModel');

const findProjects = (req, res) => {
    Projects.findProjects()
        .then(response => {
            if(response) {
                res.status(200).json(response);
            } else {
                res.status(404).json({Error: "Cannot find any projects"});
            }
        })
        .catch(err => {
            res.status(500).json({Error: err.message});
        })
}

const findProjectById = (req, res) => {
    const { id } = req.params;
    try {
        const project = Projects.findProjectById(id)
            if(project) {
                console.log(project);
                res.status(200).json(project);
            } else {
                res.status(404).json({Error: "id not found"})
            }
        
    } catch (err) {
        res.status(500).json({Error: err.message});
    }
    
}

const findProjectTasks = (req, res) => {
    const { id } = req.params;
    console.log(id)
    Projects.findProjectTasks(id)
        .then(response => {
            res.status(200).json(response);
        })
        .catch(err => {
            res.status(500).json({Error: err.message});
        })
}

const findProjectResources = (req, res) => {
    const { id } = req.params;
    console.log(id)
    Projects.findProjectResources(id)
        .then(response => {
            res.status(200).json(response);
        })
        .catch(err => {
            res.status(500).json({Error: err.message});
        })
}

const addProjects = (req, res) => {
    const { name, description, completed } = req.body;
    if(!name || !description || completed) {
        res.status(400).json({Error: "POST request must have a body"})
    }
    Projects.addProjects({name, description, completed})
        .then(response => {
            console.log("resp", response)
            res.status(201).json(response);
        })
        .catch(err => {
            res.status(500).json({Error: err.message});
        })
}

const addTasks = (req, res) => {
    const { body } = req;
    if(!body) {
        res.status(400).json({Error: "POST request must have a body"})
    }
    Projects.addProjects(body)
        .then(response => {
            res.status(201).json(response);
        })
        .catch(err => {
            res.status(500).json({Error: err.message});
        })
}

const addResources = (req, res) => {
    const { body } = req;
    const { id } = req.params;
    if(!body) {
        res.status(400).json({Error: "POST request must have a body"})
    }
    Projects.addResources(body, id)
        .then(response => {
            res.status(201).json(response);
        })
        .catch(err => {
            res.status(500).json({Error: err.message});
        })
}



router.route('/')
    .get(findProjects)
    .post(addProjects)
router.route('/:id')
    // .get(findProjectById)
router.route('/:id/tasks')
    .get(findProjectTasks)
    .post(addTasks)
router.route('/:id/resources')
    .get(findProjectResources)
    .post(addResources)






module.exports = router;






```